#!/bin/bash
#PBS -q condo
#PBS -N atac
#PBS -l nodes=1:ppn=16
#PBS -l walltime=8:00:00
#PBS -V
#PBS -m abe
#PBS -A epigen-group

#source activate bds_atac

WORKDIR="/oasis/tscc/scratch/$(whoami)/outputs/"
FASTQDIR="/projects/ps-epigen/seqdata/"

# chipseq or not
[[ "$chipseq" = true ]] &&  c="-type chip-seq"

# select libs col1:libid,col2:genome,col3:PE/SE
ncol=$(awk 'END{print NF}' $samples)
[[ $ncol -ne 3 ]] && { echo "sample file dosen't have 3 columns "; exit 0; }

samplenames=(`cat $samples`)
INPREFIX=${samplenames[${PBS_ARRAYID}*3]} #index start from 0
GENOME=${samplenames[${PBS_ARRAYID}*3+1]}
READ=${samplenames[${PBS_ARRAYID}*3+2]}

# output dir 
OUTDIR="${WORKDIR}${INPREFIX}"
mkdir -p $OUTDIR

if [ $READ = 'PE' ] ;
then
    echo "Using PE pipepline"

    # check fastqs
    fastq1="${FASTQDIR}/${INPREFIX}_R1.fastq.gz"
    fastq2="${FASTQDIR}/${INPREFIX}_R2.fastq.gz"
    echo $fastq1
    [[ ! -f $fastq1 ]] && { echo "$fastq1 fastq1 not found"; exit 0; }
    [[ ! -f $fastq2 ]] && { echo "$fastq2 fastq2 not found"; exit 0; } 

    # runPipeline
    bds /projects/ps-epigen/software/atac_dnase_pipelines/atac.bds -species $GENOME -nth $PBS_NP \
        -fastq1_1 "$fastq1" \
        -fastq1_2 "$fastq2" \
        -true_rep -no_idr  -no_par -no_xcor \
        -out_dir $OUTDIR $c
    wait

    ## addtional check
    [[ -z $(find ${OUTDIR}/qc/rep1 -name "*_qc.txt") ]]  && { echo  "pipeline failed"; exit 0; }

    # runFastQC & fastq_screen
    runFastQC_screen.sh  $INPREFIX

else
    echo "Using SE pipepline"

    # fastq check 
    fq="${FASTQDIR}/${INPREFIX}.fastq.gz"
    [[ ! -f $fq ]] && { echo "$fq fastq not found"; exit 0; }

    # run pipeline
    bds /projects/ps-epigen/software/atac_dnase_pipelines/atac.bds -species $GENOME -nth $PBS_NP \
        -se -fastq1 $fq \
        -true_rep -no_idr  -no_par \
        -out_dir $OUTDIR $c
    wait

    ## addtional check
    [[ -z $(find ${OUTDIR}/qc/rep1 -name "*_qc.txt") ]]  && { echo  "pipeline failed"; exit 0; }
    
    # fastqc and screen
    runFastQC_screen_se.sh $INPREFIX 
fi


# result transfer 
results_transfer.sh $INPREFIX $(whoami)

# add .finished.txt tag
touch /projects/ps-epigen/outputs/libQCs/${INPREFIX}/.finished.txt



